### DISCUSSION TITLE
How to add support for a new database dialect?

### DISCUSSION THREAD

**Discussion by clamx4** [sequelize/sequelize#16939] [Help & Questions] [answered]
I would like to contribute to Sequelize by adding support for a new database.

I notice that there are issues like https://github.com/sequelize/sequelize/issues/5984 and https://github.com/sequelize/sequelize/issues/13291 asking for support for HANA. Currently HANA is not in the available database list. It would be great if Sequelize can support HANA.

I have read the CONTRIBUTING.md but I didn't find how to add an new database.
Is there any documentation or guidelines to add support for a new database that I can refer to?

**Comment by WikiRik** [comment, marked as answer]
We don't have good documentation on how to add a new dialect since it does not happen that often. In the last two years we've added support for snowflake and ibmi but we have added a warning for those because we cannot run integration tests. 

One open PR for a new dialect where we can run the integration tests is https://github.com/sequelize/sequelize/pull/15569 but I think we would recommend making a fork and adding support for a new dialect in there and wait for sequelize v8 in which we'll add a plugin system for supporting dialects and you can easily make a community plugin that adds support for HANA.

**Reply by clamx4** [reply]
Thank you WikiRik.
The PR https://github.com/sequelize/sequelize/pull/15569 for adding CockroachDb support is a great reference for me.

And just a remark: https://github.com/orgs/sequelize/projects/1/views/1, where the 'Split dialects in different modules' is listed in the roadmap.
